homeassistant/components/august/sensor.py

```diff
@@ -70,9 +86,111 @@ def _retrieve_linked_keypad_battery_state(detail):
                 )
                 devices.append(AugustBatterySensor(data, sensor_type, device))
 
+    for device in operation_sensors:
+        devices.append(AugustOperatorSensor(data, device))
+
     async_add_entities(devices, True)
 
 
+class AugustOperatorSensor(AugustEntityMixin, RestoreEntity, Entity):
+    """Representation of an August lock operation sensor."""
+
+    def __init__(self, data, device):
+        """Initialize the sensor."""
+        super().__init__(data, device)
+        self._data = data
+        self._device = device
+        self._state = None
+        self._operated_remote = None
+        self._operated_keypad = None
+        self._operated_autorelock = None
+        self._operated_time = None
+        self._available = False
+        self._entity_picture = None
+        self._update_from_data()
+
+    @property
+    def available(self):
+        """Return the availability of this sensor."""
+        return self._available
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self._device.device_name} Operator"
+
+    @callback
+    def _update_from_data(self):
+        """Get the latest state of the sensor and update activity."""
+        lock_activity = self._data.activity_stream.get_latest_device_activity(
+            self._device_id, [ActivityType.LOCK_OPERATION]
+        )
+
+        if lock_activity is not None:
+            self._available = True
+            self._state = lock_activity.operated_by
```

Comment:
Isn't this the same info as the lock?

Reply:
Yes, the goal here is to provide a nice state history in the UI and the ability to do Presence Detection with lock operations. 
https://deploy-preview-12314--home-assistant-docs.netlify.com/integrations/august/

![Uploading Screen Shot 2020-03-09 at 4.20.23 PM.png…]()

<img width="745" alt="Screen Shot 2020-03-09 at 4 21 48 PM" src="https://user-images.githubusercontent.com/663432/76258715-1f31f800-6222-11ea-917e-4edb7a6a399e.png">

Reply:
I'm thinking I should remove the new attributes from the lock and focus the user on the sensor.

Reply:
I've done that

Reply:
> I've done that

Did you forget to push it?

Reply:
Pushed to august_lock_operation_sensors doh.

fixing

Reply:
@balloob  Fixed up the conflict. sorry about that

Reply:
Manually diffed to my test branch as well to verify I didn't mess up the resolution. and retested basic functionality